crates/spark/src/services/coop_exit.rs

```diff
@@ -177,19 +181,29 @@ impl CoopExitService {
 
         // Request cooperative exit from the SSP
         trace!("Requesting cooperative exit");
-        let coop_exit_request = self
+        let res = self
             .ssp_client
             .request_coop_exit(RequestCoopExitInput {
                 leaf_external_ids,
                 withdrawal_address: withdrawal_address.to_string(),
-                idempotency_key: None,
+                idempotency_key: Some(unwrapped_transfer_id.to_string()),
                 exit_speed: exit_speed.into(),
                 withdraw_all,
                 fee_leaf_external_ids,
                 fee_quote_id,
-                user_outbound_transfer_external_id: Some(transfer_id.to_string()),
+                user_outbound_transfer_external_id: None,
             })
```

Comment:
Does this actually set the transfer id of the resulting transfer now? If not, I think we should rename the transfer_id field here and in upper layers to idempotency_key instead.